checks/gcp_incidents.go

```diff
@@ -0,0 +1,256 @@
+	ID           string `json:"incident_id,omitempty"`
+	Summary      string `json:"summary,omitempty"`
+	State        string `json:"state,omitempty"`
+	ResourceName string `json:"resource_name,omitempty"`
+
+	m map[string]any
+}
+
+func (g *GCPIncident) ToMapStringAny() map[string]any {
+	if g.m == nil {
+		b, _ := json.Marshal(g)
+		_ = json.Unmarshal(b, &g.m)
+	}
+	return g.m
+}
+
+type GCPIncidentList struct {
+	Incidents []GCPIncident `json:"incidents,omitempty"`
+}
+
+type GCPIncidentPayload struct {
+	Incident GCPIncident `json:"incident,omitempty"`
+}
+
+func CheckGCPIncidents(ctx *context.Context, extConfig external.Check) pkg.Results {
+	var results pkg.Results
+	pubSubCheck := extConfig.(v1.PubSubCheck)
+	check := pubSubCheck.GCPIncidents
+	result := pkg.Success(pubSubCheck, ctx.Canary)
+	results = append(results, result)
+
+	var tokenSrc oauth2.TokenSource
+	if check.ConnectionName != "" {
+		err := check.GCPConnection.HydrateConnection(ctx)
+		if err != nil {
+			return results.ErrorMessage(fmt.Errorf("error hydrating connection %s: %w", check.ConnectionName, err))
+		}
+		tokenSrc, err = check.GCPConnection.TokenSource(ctx.Context)
+		if err != nil {
+			return results.ErrorMessage(fmt.Errorf("error getting token source for %s/%s: %w", check.Project, check.Subscription, err))
+		}
+	} else {
+		creds, err := gcp.DefaultCredentials(ctx)
+		if err != nil {
+			return results.ErrorMessage(fmt.Errorf("error creating default creds for %s/%s: %w", check.Project, check.Subscription, err))
+		}
+		tokenSrc = creds.TokenSource
+	}
+
+	conn, cleanup, err := gcppubsub.Dial(ctx, tokenSrc)
+	if err != nil {
+		return results.ErrorMessage(fmt.Errorf("error connecting to GCP: %w", err))
+	}
+	defer cleanup()
+
+	subClient, err := gcppubsub.SubscriberClient(ctx, conn)
+	if err != nil {
+		return results.ErrorMessage(fmt.Errorf("error opening subscription for %s/%s: %w", check.Project, check.Subscription, err))
+	}
+	defer subClient.Close()
+
+	subscription, err := gcppubsub.OpenSubscriptionByPath(subClient, fmt.Sprintf("projects/%s/subscriptions/%s", check.Project, check.Subscription), nil)
+	if err != nil {
+		return results.ErrorMessage(fmt.Errorf("error opening subscription for %s/%s: %w", check.Project, check.Subscription, err))
+	}
+	//nolint:errcheck
+	defer subscription.Shutdown(ctx)
+
+	msgs, err := ListenWithTimeout(ctx, subscription, 10*time.Second)
+	if err != nil {
+		return results.ErrorMessage(fmt.Errorf("error listening to subscription for %s/%s: %w", check.Project, check.Subscription, err))
+	}
+
+	var pubSubResult PubSubResults
+	for _, rawMsg := range msgs {
+		var payload GCPIncidentPayload
+		_ = json.Unmarshal([]byte(rawMsg), &payload)
+
+		inc := payload.Incident
+		// We ignore acknowledgement state
+		switch inc.State {
+		case "closed":
+			// We set the status to true directly in database
+			if ctx.DB() != nil {
+				if pubSubCheck.ResultLookup == "" {
+					pubSubCheck.ResultLookup = `$.incident_id`
```

Comment:
this should be in the cel function

Reply:
I think it might also make sense todo `stringToUUIDv5(incidentId)`  that we can use then use as a check_id